monash-human-power/MHP_DAS
# Add Battery voltage reader class

## Description

This pull request adds the `BatteryReader` class to the WM's arsenal to read the battery voltage powering the WM. The other relevant files have been updated to reflect this new entry. 

This makes our middle WM (or at least the MVP) complete in that it can carry out all the significant tasks required for it to conduct including reading data from the CO2, accelerometer, gyroscope, temperature and humidity sensors and reading battery voltage and sending them wirelessly using `mqtt`.

## Screenshots
Note I didn't have a battery connected in this case:
<img width="1225" alt="Screen Shot 2021-01-09 at 6 27 52 pm" src="https://user-images.githubusercontent.com/63642262/104086050-783a4480-52a8-11eb-8d8d-b86043201afa.png">

## Steps to Test

1) Connect ESP32 to your computer or to test the battery voltage algorithm only use a battery to power the board.
2) The board should automatically run `boot.py` and then `main.py`
3) Use terminal window and enter `mosquitto_pub -h <broker name> -t /v3/wireless_module/2/start -m ""` to inform ESP32 to start sending data (Remember your `broker name` should match the one in your `config.py`)
4) Use `mosquitto_sub -h <broker name> -t /v3/wireless_module/2/data` to read sensor data and ensure they run correctly as they did before this PR.
5) Use `mosquitto_sub -h <broker name> -t /v3/wireless_module/2/battery` to read battery voltage. Remember that it only publishes battery data every 5 minutes. If needed for testing, you can change this rate in `main.py` or `wireless_module.py`.
6) Check that the data published confirms with your own reading using a voltmeter.
